Preproc (#688)
mlr-org/mlr3book: book/chapters/chapter9/preprocessing.qmd

Alternatively, one can impute by sampling from the empirical distribution of the feature, for example a histogram (`po("imputehist")`).
Instead of guessing at what a missing feature might be, missing values could instead be replaced by a new level, for example, called `.MISSING` (`po("imputeoor")`).
For numeric features, @ding2010investigation show that for binary classification and tree-based models, encoding missing values out-of-range (OOR), e.g. a constant value above the largest observed value, is a reasonable approach.

```{r, echo = FALSE, out.width = "60%"}
#| label: fig-imputation
#| fig-cap: Mean imputation of missing values using observed values.
#| fig-alt: "On the left is a vector of numbers in a column, (1.3, NA, 1.5, NA). The non-NA numbers have arrows pointing to (1.3+1.5)/2, which then has an arrow pointing to a vector of numbers in a column on the right but now (1.3, 1.4, 1.5, 1.4) with '1.4' in red to highlight they were imputed with the mean."
include_multi_graphics("mlr3book_figures-13")
```

> It is very important for predictive tasks that you keep track of missing data as it is common for missing data to be informative in itself.

Reviewer: ```suggestion
It is very important for predictive tasks that you keep track of missing data as it is common for the reasons behind missing data to be informative.
```

> To preserve the information about the missingness for the model, imputation should be tracked by adding binary indicator features (one for each imputed feature) that are `1` if the feature was missing for an observation and `0` if it was present (`po("missind")`).

Reviewer: ```suggestion
To preserve information about missing data, imputation should be tracked by adding binary indicator features (one for each imputed feature) that are `1` if the feature was missing for an observation and `0` if it was present (`po("missind")`).
```

> But also problems in model interpretation can arise, as in this real-world example: medical data are typically collected more extensively for white communities than for racially minoritized communities.
> Imputing data from minoritized communities would at best mask this data bias, and at worst would make the data bias even worse by making vastly inaccurate assumptions (see @sec-fairness for data bias and algorithmic fairness).

Reviewer: ```suggestion
It is important to note that recording this information may not prevent problems in model interpretation.
As a real-world example, medical data are typically collected more extensively for White communities than for racially minoritized communities.
Imputing data from minoritized communities would at best mask this data bias, and at worst would make the data bias even worse by making vastly inaccurate assumptions (see @sec-fairness for data bias and algorithmic fairness).
```

Alternatively, one can impute by sampling from the empirical distribution of the feature, for example a histogram (`po("imputehist")`).
Instead of guessing at what a missing feature might be, missing values could instead be replaced by a new level, for example, called `.MISSING` (`po("imputeoor")`).
For numeric features, @ding2010investigation show that for binary classification and tree-based models, encoding missing values out-of-range (OOR), e.g. a constant value above the largest observed value, is a reasonable approach.

```{r, echo = FALSE, out.width = "60%"}
#| label: fig-imputation
#| fig-cap: Mean imputation of missing values using observed values.
#| fig-alt: "On the left is a vector of numbers in a column, (1.3, NA, 1.5, NA). The non-NA numbers have arrows pointing to (1.3+1.5)/2, which then has an arrow pointing to a vector of numbers in a column on the right but now (1.3, 1.4, 1.5, 1.4) with '1.4' in red to highlight they were imputed with the mean."
include_multi_graphics("mlr3book_figures-13")
```

> It is very important for predictive tasks that you keep track of missing data as it is common for the reasons behind missing data to be informative.

Reviewer: ```suggestion
It is essential for predictive tasks that you keep track of missing data as it is common for the reasons behind missing data to be informative.
```

To preserve information about missing data, imputation should be tracked by adding binary indicator features (one for each imputed feature) that are `1` if the feature was missing for an observation and `0` if it was present (`po("missind")`).

> It is important to note that recording this information may not prevent problems in model interpretation.

Reviewer: ```suggestion
It is important to note that recording this information will not prevent problems in model interpretation on its own.
```

```

Similarly to encoding, we see limited differences in performance between the different imputation strategies.

> This is expected here and confirms the findings of @ding2010investigation - out-of-range imputation is a simple yet effective imputation for tree-based methods.

Reviewer: ```suggestion
This is expected here and confirms the findings of @ding2010investigation -- out-of-range imputation is a simple yet effective imputation for tree-based methods.
```

Many more advanced imputation strategies exist, including model-based imputation where machine learning models are used to predict missing values, and multiple imputation where data is repeatedly resampled and imputed in each sample (e.g., by mean imputation) to attain more robust estimates.
However, these more advanced techniques rarely improve the models predictive performance substantially and the simple imputation techniques introduced above are usually sufficient (@Poulos2018).

> Nevertheless, these methods have their justification: Not for the predictive performance, but for finding imputations that fit well to the distribution of the observed values, so that a model can be fitted that can be interpreted and analyzed in a second step.

Reviewer: ```suggestion
Nevertheless, these methods are still important, as finding imputations that fit well to the distribution of the observed values allows a model to be fitted that can be interpreted and analyzed in a second step.
```
